wifimanager_adapter.cpp

```diff
@@ -0,0 +1,136 @@
+#include "wifimanager_adapter.h"
+
+void WifiManagerAdapter::setup(Configuration *config)
```

Comment:
could this function be broken up into a couple different functions that describe the steps?